core/common/src/main/java/alluxio/util/JvmPauseMonitor.java

```diff
@@ -228,4 +233,59 @@ public void run() {
       }
     }
   }
+
+  /**
+   * An unmodifiable view of a garbage collector MX Bean.
+   */
+  public static class GarbageCollectorMXBeanView implements GarbageCollectorMXBean {
```

Comment:
I don't fully understand why this class is needed

Also, wouldn't it be more memory efficient (and easier to code) to just hold a reference to the bean, and call the methods themselves instead of storing all these variables?

Reply:
If you run this code
```java
public static void main(String[] args) {
  List<GarbageCollectorMXBean> oldBean = ManagementFactory.getGarbageCollectorMXBeans();
  System.out.println(oldBean);
  System.out.println("old bean info:");
  oldBean.forEach(it -> System.out.println(formatGcBean(it)));
  System.gc();
  System.out.println("===== system gc =====");
  List<GarbageCollectorMXBean> newBean = ManagementFactory.getGarbageCollectorMXBeans();
  System.out.println(newBean);
  System.out.println("old bean info:");
  oldBean.forEach(it -> System.out.println(formatGcBean(it)));
  System.out.println("new bean info:");
  newBean.forEach(it -> System.out.println(formatGcBean(it)));
}

private static String formatGcBean(GarbageCollectorMXBean it) {
  return "name:" + it.getName() + " count:" + it.getCollectionCount();
}
```
and the result is 
```
[sun.management.GarbageCollectorImpl@378fd1ac, sun.management.GarbageCollectorImpl@49097b5d]
old bean info:
name:PS Scavenge count:0
name:PS MarkSweep count:0
===== system gc =====
[sun.management.GarbageCollectorImpl@378fd1ac, sun.management.GarbageCollectorImpl@49097b5d]
old bean info:
name:PS Scavenge count:1
name:PS MarkSweep count:1
new bean info:
name:PS Scavenge count:1
name:PS MarkSweep count:1
```
so you can see `ManagementFactory.getGarbageCollectorMXBeans()` will return the same gc bean instance, so I need this class to store previous data.